scripts/gha/build_testapps.py

```diff
@@ -550,21 +551,25 @@ def patch_android_env(unity_version):
       logging.info(str(e))
 
   try:
-    # The platform android-33 includes libraries that were built with Java 11, and require a newer version of gradle
-    # than Unity comes with. Note this only happens when using minification.
-    # If this continues to be a problem, this logic might need to be smarter, to remove all versions newer than 32,
-    # but currently the GitHub runners have 33 as their max.
-    logging.info("Uninstall Android platform android-33")
-    _run([sdkmanager_path, "--uninstall",
-          "platforms;android-33", "platforms;android-33-ext4", "platforms;android-33-ext5",
-          "platforms;android-34"], check=False)
+    # The platform android-33 includes libraries that were built with Java 11,
+    # and require a newer version of gradle than Unity comes with. Note this
+    # only happens when using minification.
+    list_installed_result = _run(
+        [sdkmanager_path, "--list"], capture_output=True, text=True, check=False)
+    # This should match both platforms;android-33 and platforms;android-33-ext4.
+    matches = set(re.findall(
+        r'(platforms;android-(\d+)-?\w*)', list_installed_result.stdout))
+    installed_versions = [
+        platform for platform, version in matches if int(version) >= 33]
+    if installed_versions:
+      _run([sdkmanager_path, "--uninstall"] + installed_versions, check=False)
   except Exception as e:
-    logging.exception("Failed to uninstall Android platform android-33")
-
+    logging.exception("Failed to uninstall Android platforms;android-33")
   try:
-    # List the installed packages to make it easier to notice if any incompatible packages are present.
+    # List the installed packages to make it easier to debug if any incompatible
+    # packages are present.
     logging.info("Listing installed sdks")
-    _run([sdkmanager_path, "--list_installed"], check=False)
+    _run([sdkmanager_path, "--list"], check=False)
```

Comment:
What is the difference between list and list_installed?

Reply:
My understanding is that list_installed is only an option for the windows sdkmanager.bat, but list is supported by both mac and windows sdkmanagers. This page mentions 'list' but has no mention of 'list_installed':
https://developer.android.com/tools/sdkmanager#list
I switched to 'list' as it should work even if we switched the android build to happen on mac or linux instead of windows.

list_installed outputs a list of installed packages. list outputs the same list of installed packages followed by an additional list of 'available' packages. This does mean that it can result in 'false positives' where it will attempt to uninstall an available package, but I verified that the uninstall command succeeds even if it attempts to uninstall something that was not already installed.

You can see an example of the 'list' output in the 'Build integration tests' step of this run:
https://github.com/firebase/firebase-unity-sdk/actions/runs/5360532633/jobs/9725527898

Reply:
Cool, as long as there is some way to look at the logs and see what is installed vs not, that sounds good.